resources/lib/functions.py

```diff
@@ -134,6 +134,25 @@ def mainEntryPoint():
         checkService()
         checkServer(notify=False)
         showContent(sys.argv[0], int(sys.argv[1]), params)
+    elif mode == "SEARCH":
+        # plugin://plugin.video.embycon?mode=SEARCH
+        checkService()
+        checkServer(notify=False)
+        xbmcplugin.setContent(int(sys.argv[1]), 'files')
+        showSearch()
+    elif mode == "NEW_SEARCH":
+        # plugin://plugin.video.embycon?mode=NEW_SEARCH&item_type=<Movie|Series|Episode>
+        if 'SEARCH_RESULTS' not in xbmc.getInfoLabel('Container.FolderPath'):  # don't ask for input on '..'
```

Comment:
This way if Kodi causes the page to refresh on results it won't prompt for input, it will just refresh the page. And the '..' brings you back to the Search sub-menu to select Movie/Series/Episode to search again.